package.json

```
@@ -0,0 +1,41 @@
+{
+  "name": "crud-ts",
+  "version": "1.0.0",
+  "description": "This application allows user to create new product, update existing product and user is able to filter product by product id and able to sort it by productId, Product Name and Price.",
+  "type": "module",
+  "scripts": {
+    "dev": "parcel serve ./src/index.html --port 3000",
+    "type-check": "tsc -b --noEmit",
+    "lint": "eslint src --ext .ts",
+    "build": "npm run type-check && npm run lint && parcel build ./src/index.html --no-autoinstall --no-optimize",
+    "format": "prettier src --write"
```

It is a good idea to first run the format before running the lint. We can potentially avoid the conflicts if any.

```suggestion
    "format": "prettier src --write",
    "build": "npm run format && npm run type-check && npm run lint && parcel build ./src/index.html --no-autoinstall --no-optimize",
```

---

I don't think we need to format at build time. Because this build script will run on Netlify and we only care about resultant bundles for deployment. And formatting at build time will not change the output bundles and we will introduce some overhead by formatting at build time.

---

Yes, that is a valid point. 
However, what I suggested that we can make sure, there will be not such error we encounter due to lack of formatting. It is not necessary if we are formatting the code and prettifying it earlier. Just my two cents.

---

I am using prettier extension in VS Code. It formats file whenever I save it. So I think all files will be already formatted at built time.

---

Okay, in that case it's fine.